From b5d17894e396b9b9fcd42fa4c5ef45e3c54f0f19 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Mon, 17 Jun 2013 10:46:00 +0900
Subject: [PATCH 502/715] drm: rcar-du: Add signal selection support

Output of DU0 can only select Analog RGB connector.
Output of DU1 can select Analog RGB connector or LVDS connector.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
(cherry picked from commit 0c678c357f4037423e7974178695fcc9d3da6d13)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/Kconfig       |   21 +++++++++++++++++++++
 drivers/gpu/drm/rcar-du/rcar_du_drv.c |    7 +++++++
 2 files changed, 28 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/Kconfig b/drivers/gpu/drm/rcar-du/Kconfig
index 2eb7d23..0566ef1 100644
--- a/drivers/gpu/drm/rcar-du/Kconfig
+++ b/drivers/gpu/drm/rcar-du/Kconfig
@@ -7,3 +7,24 @@ config DRM_RCAR_DU
 	help
 	  Choose this option if you have an R-Car chipset.
 	  If M is selected the module will be called rcar-du-drm.
+
+config ANALOG_RGB_CH
+	int "Display analog RGB output channel"
+	range 0 1
+	default 0
+	depends on DRM_RCAR_DU
+	help
+	  This option is selection of DU channel for analog RGB
+	  output.
+	  if you want to set Analog RGB output to DU0, you set "0".
+	  if you want to set Analog RGB output to DU1, you set "1".
+
+config LVDS_CH
+	int "Display LVDS output channel"
+	range 1 1
+	default 1
+	depends on DRM_RCAR_DU
+	help
+	  This option is selection of DU channel for LVDS
+	  output.
+	  if you want to set LVDS output to DU1, you set "1".
diff --git a/drivers/gpu/drm/rcar-du/rcar_du_drv.c b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
index e457598..161913c 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_drv.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
@@ -74,6 +74,13 @@ int rcar_du_get(struct rcar_du_device *rcdu)
 	rcar_du_write(rcdu, DEFR4, DEFR4_CODE);
 	rcar_du_write(rcdu, DEFR5, DEFR5_CODE | DEFR5_DEFE5);
 
+	if (CONFIG_ANALOG_RGB_CH == 2)
+		rcar_du_write(rcdu, DEFR8, DEFR8_CODE | DEFR8_DRGBS_DU2);
+	else if (CONFIG_ANALOG_RGB_CH == 1)
+		rcar_du_write(rcdu, DEFR8, DEFR8_CODE | DEFR8_DRGBS_DU1);
+	else
+		rcar_du_write(rcdu, DEFR8, DEFR8_CODE | DEFR8_DRGBS_DU0);
+
 	/* Use DS1PR and DS2PR to configure planes priorities and connects the
 	 * superposition 0 to DU0 pins. DU1 pins will be configured dynamically.
 	 */
-- 
1.7.10.4

